providers/os/detector/detector_all.go

```diff
@@ -703,36 +704,83 @@ var windows = &PlatformResolver{
 	Name:     "windows",
 	IsFamily: false,
 	Detect: func(r *PlatformResolver, pf *inventory.Platform, conn shared.Connection) (bool, error) {
-		data, err := win.GetWmiInformation(conn)
-		if err != nil {
-			log.Debug().Err(err).Msg("could not gather wmi information")
-			return false, nil
-		}
+		if conn.Capabilities().Has(shared.Capability_RunCommand) {
+			data, err := win.GetWmiInformation(conn)
```

Comment:
I wonder if we extract the individual detections (runtime and static) into to functions so the detect function is easier to read?